oxidation/libparsec/crates/types/src/regex.rs

```diff
@@ -0,0 +1,35 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+use std::str::FromStr;
+
+use fancy_regex::{self, RegexBuilder};
+
+pub struct Regex(pub fancy_regex::Regex);
+
+// The fnmatch_regex crate does not escape the character '$'. It is a problem
+// for us. Since we need to match strings like `$RECYCLE.BIN` we need to escape
+// it anyway.
+fn escape_dollarsign(string: &str) -> String {
+    str::replace(string, "$", "\\$")
+}
```

Comment:
So this fonction will also escape the trailing `$` that represent the `end of line`. 
This look like an edge case that need to be handled

Reply:
Oh, then
WDYT of if the function was named `escape_globing_pattern` ?